Content.Server/GameObjects/Components/Markers/TrashSpawnerComponent.cs

```diff
@@ -0,0 +1,87 @@
+using System;
+using Content.Server.GameObjects.Components.Markers;
+using System.Collections.Generic;
+using Content.Server.GameTicking;
+using Content.Server.Interfaces.GameTicking;
+using Robust.Server.Interfaces.GameObjects;
+using Robust.Shared.GameObjects;
+using Robust.Shared.Interfaces.GameObjects;
+using Robust.Shared.Interfaces.Random;
+using Robust.Shared.Maths;
+using Robust.Shared.IoC;
+using Robust.Shared.Random;
+using Robust.Shared.Serialization;
+using Robust.Shared.ViewVariables;
+using Logger = Robust.Shared.Log.Logger;
+
+namespace Content.Server.GameObjects.Components.Markers
+{
+    [RegisterComponent]
+    public class TrashSpawnerComponent : ConditionalSpawnerComponent
+    {
+        public override string Name => "TrashSpawner";
+
+#pragma warning disable 649
+        [Dependency] private IEntityManager _entityManager;
+        [Dependency] private IRobustRandom _robustRandom;
+#pragma warning restore 649
+
+        [ViewVariables(VVAccess.ReadWrite)]
+        public List<string> RarePrototypes { get; set; } = new List<string>();
+
+        [ViewVariables(VVAccess.ReadWrite)]
+        private List<string> _gameRules = new List<string>();
+
+        [ViewVariables(VVAccess.ReadWrite)]
+        public float RareChance { get; set; } = 0.05f;
+
+        [ViewVariables(VVAccess.ReadWrite)]
+        public float Offset { get; set; } = 0.2f;
+
+        public override void ExposeData(ObjectSerializer serializer)
+        {
+            base.ExposeData(serializer);
+            serializer.DataField(this, x => RarePrototypes, "rare_prototypes", new List<string>());
+            serializer.DataField(this, x => RareChance, "rare_chance", 0.05f);
```

Comment:
These names should follow the [YAML naming guidelines](https://hackmd.io/@ss14/code-quality-control).

>### Naming
>`PascalCase` is used for IDs and component names. Everything else uses `camelCase`.